src/main.rs

```diff
@@ -57,7 +58,19 @@ fn run_target_program(elf_path: &Path, chip_name: &str, opts: &cli::Opts) -> any
     let mut sess = if opts.connect_under_reset {
         probe.attach_under_reset(probe_target)?
     } else {
-        probe.attach(probe_target)?
+        let probe_attach = probe.attach(probe_target);
+        if let Err(probe_rs::Error::Probe(ProbeSpecific(e))) = &probe_attach {
+            if e.to_string().contains("JtagNoDeviceConnected") {
+                eprintln!("Info: Jtag cannot find a connected device.");
+                eprintln!("Help:");
+                eprintln!("    Try using probe-run with option `--connect-under-reset`");
+                eprintln!("    Or if using cargo:");
+                eprintln!("        cargo run -- --connect-under-reset");
+                eprintln!("    This error comes from the program currently in the chip and");
```

Comment:
We shouldn't state this with certainty, it is possible that there really just isn't a device connected

Reply:
Maybe we should add something like "Check that the debugger is connected to the chip, if so try […]" and then "If this flag fixed your issue, this error might come from […]"?

Reply:
> "Check that the debugger is connected to the chip, if so try […]" and then "If this flag fixed your issue, this error might come from […]"?

That sounds good